server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/disk/schemafile/SchemaFile.java

```diff
@@ -648,90 +706,29 @@ private boolean isStorageGroupNode(IMNode node) {
   // endregion
   // region Segment Address Operation
 
-  /**
-   * It might be a weird but influential method. It searches for the real segment address with
-   * address from parental node and child name.
-   *
-   * <p>For now, it examines segments one by one with target child name, which has Operation(n).
-   *
-   * @param curSegAddr segment address from parental node
-   * @param recKey target child name
-   */
   private long getTargetSegmentAddress(long curSegAddr, String recKey)
       throws IOException, MetadataException {
-    // TODO: improve efficiency
-    short curSegId = getSegIndex(curSegAddr);
-
+    treeTrace.get()[0] = 0;
     ISchemaPage curPage = getPageInstance(getPageIndex(curSegAddr));
-
-    if (curPage.hasRecordKeyInSegment(recKey, curSegId)) {
+    if (curPage.getAsSegmentedPage() != null) {
       return curSegAddr;
     }
 
-    long nextSegAddr = curPage.getNextSegAddress(curSegId);
-    while (nextSegAddr >= 0) {
-      ISchemaPage pivotPage = getPageInstance(getPageIndex(nextSegAddr));
-      short pSegId = getSegIndex(nextSegAddr);
-      if (pivotPage.hasRecordKeyInSegment(recKey, pSegId)) {
-        return nextSegAddr;
-      }
-      nextSegAddr = pivotPage.getNextSegAddress(pSegId);
+    int i = 0; // mark the trace of b+ tree node
+    while (curPage.getAsInternalPage() != null) {
+      i++;
+      treeTrace.get()[i] = curPage.getPageIndex();
+      curPage = getPageInstance(curPage.getAsInternalPage().getRecordByKey(recKey));
```

Comment:
It seems the threadLocal ```treeTrace``` is used for the thread's re-entrance operation. From the development efficiency, the application is fine. For more elegant implementation, we can redesign and refactor the interaction of MTree, MTreeStore and SchemaFile.

Reply:
As mentioned above, ThreadLocal is no longer involved.